test/unit/test_borealis.py

```diff
@@ -0,0 +1,398 @@
+# Copyright 2019 SuperDARN Canada, University of Saskatchewan
+# Author: Marci Detwiller
+"""
+This test suite is to test the implementation for the following classes:
+    BorealisRead
+    BorealisUtilities
+    BorealisWrite
+Support for the following Borealis file types:
+    rawrf
+    antennas_iq
+    bfiq
+    rawacf
+And supports conversion to the following Borealis types:
+    bfiq -> iqdat
+    rawacf -> rawacf
+"""
+
+import unittest
+import numpy as np
+import logging
+import collections
+import bz2
+import os
+import copy
+
+import pydarn
+import borealis_rawacf_data_sets
+import borealis_bfiq_data_sets
+
+pydarn_logger = logging.getLogger('pydarn')
+
+# Test files
+
+borealis_bfiq_file = "../test_files/20190523.0200.00.sas.0.bfiq.hdf5"
+borealis_rawacf_file = "../test_files/20190523.0200.00.sas.0.rawacf.hdf5"
+# borealis_antennas_iq_file = 
+# borealis_rawrf_file =
+
+# Problem files
+
+
+class TestBorealisRead(unittest.TestCase):
+    """
+    Testing class for BorealisRead
+    """
+
+    def setUp(self):
+        pass
+
+    def test_incorrect_path(self):
+        """
+        Testing BorealisRead constructor with an nonexistent folder.
+
+        Expected behaviour: raise OSError
+        """
+        self.assertRaises(OSError, pydarn.BorealisRead,
+                          './dog/somefile.rawacf')
+
+    def test_incorrect_file(self):
+        """
+        Tests if BorealisRead constructor with an non-existent file
+
+        Expected behaviour: raises OSError
+        """
+        self.assertRaises(OSError, pydarn.BorealisRead,
+                          '../test_files/somefile.rawacf')
+
+    def test_empty_file(self):
+        """
+        Tests if BorealisRead constructor with an empty file
+
+        Expected behaviour: raise OSError, unable to open (file signature not found)
+        """
+        self.assertRaises(OSError,
+                          pydarn.BorealisRead, '../test_files/empty.rawacf')
+
+    def test_open_rawacf(self):
+        """
+        Tests BorealisRead constructor on opening a rawacf.
+        It should be able to open the file, read it and convert to dictionary
+
+        Checks:
+            - group names is a list, greater than length 0
+        """
+        file_path = borealis_rawacf_file
```

Comment:
Maybe as an enhancement/refactors of code you can put file_path in setUp thus letting you call self.file_path and removing that 1 line of code that is repeating in the test functions.

Reply:
done!